File: helpdesk/src/Helpdesk.Models/Issue/IssueCommentModel.cs
using System.Collections.Generic;

namespace Helpdesk.Models.Issue
{
    public class IssueCommentModel
    {
        public int IssueId { get; set; }
        public string Comment { get; set; }
        public IEnumerable<DocumentModel> Documents { get; set; }
    }
}

File: helpdesk/src/Helpdesk.SPA/src/models/issueCommentModel.js
import { DocumentModel } from './documentModel.js';
export class IssueCommentModel {

  constructor(obj = {}) {
    this.id = obj.id || null;
    this.issueId = obj.issueId;
    this.comment = obj.comment;
    this.documents = obj.documents ? obj.documents.map(el => new DocumentModel(el)) : [];
  }
}
